### PR TITLE
Feat: add goerli

### LINKED ISSUE #6441: Add Goerli testnet to Trezor Suite
At the end of the year Ropsten testnet will be deprecated and we need to replace it.
Goerli testnet needs to be introduced in Trezor Suite.
https://goerli1.trezor.io/
https://goerli2.trezor.io/
Do not remove Ropsten yet.
Goerli testnet token needs to follow the same user flow as Ropsten -> add account in Testnet coins section.

### PR DESCRIPTION
**Pull request by karliatto**
## Description
Adds Goerli testnet support for trezor-suite.

The script `scripts/update-coins.sh` has being doing more changes to the file `coins.json`. I assume that those were coins that were for some reason deleted from the firmware and now when updating suite have being removed, but I am not 100% sure.

The logo for goerli I have used the same as for Ethereum Ropsten Testnet.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/6441

### CODE REVIEW

File: packages/components/src/components/logos/CoinLogo/coins.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export const COINS = {
     nmc: require('../../../images/coins/nmc.svg'),
     tada: require('../../../images/coins/tada.svg'),
     trop: require('../../../images/coins/trop.svg'),
+    tgor: require('../../../images/coins/trop.svg'),
```
Comment (karliatto) [inline packages/components/src/components/logos/CoinLogo/coins.ts:18]: I have used the same logo that we have for Ropsten, but not sure if we should have any different one.

Asked for it in the issue https://github.com/trezor/trezor-suite/issues/6441#issuecomment-1283570786

File: packages/connect-common/files/coins.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
             "min_address_length": 27,
             "minfee_kb": 1000,
             "name": "Bitcoin",
+            "overwintered": false,
```
Comment (karliatto) [inline packages/connect-common/files/coins.json:41]: I have not modified this file manually, only with the script `scripts/update-coins.sh`. I guess those `"overwintered": false` are new thing or the script needs to be fixed? @mroz22

File: packages/connect-common/files/coins.json [hunk 3]
```diff
@@ -1068,54 +1090,6 @@
             "xpub_magic_segwit_native": 78792518,
             "xpub_magic_segwit_p2sh": 77429938
         },
-        {
-            "address_type": 38,
-            "address_type_p2sh": 62,
-            "bech32_prefix": "game",
-            "blockchain_link": {
```
Comment (karliatto) [inline packages/connect-common/files/coins.json:1075]: This object is being deleted by running the script `scripts/update-coins.sh`. I assume this is not in the firmware anymore and therefore it is deleted. @mroz22

File: packages/connect-common/files/coins.json [hunk 4]
```diff
@@ -1404,51 +1384,6 @@
             "xpub_magic_segwit_native": null,
             "xpub_magic_segwit_p2sh": null
         },
-        {
-            "address_type": 38,
-            "address_type_p2sh": 53,
-            "bech32_prefix": "nix",
-            "blockchain_link": null,
-            "blocktime_seconds": 120,
```
Comment (karliatto) [inline packages/connect-common/files/coins.json:1412]: Same as before, this was deleted. I am not sure if this is the correct behavior of the script `scripts/update-coins.sh`.

File: packages/connect-common/files/coins.json (hunk 2 above)
Comment (mroz22) [reply packages/connect-common/files/coins.json:41]: this is ok, they probably updated definitions in trezor-firwmare. overwintered applies only for zcash, other coins have it either undefined or set to false.

File: packages/connect-common/files/coins.json (hunk 3 above)
Comment (mroz22) [reply packages/connect-common/files/coins.json:1075]: yes, that should be ok

File: packages/connect-common/files/coins.json [hunk 5]
```diff
@@ -1593,54 +1531,6 @@
             "xpub_magic_segwit_native": 73342198,
             "xpub_magic_segwit_p2sh": 71979618
         },
-        {
-            "address_type": 55,
-            "address_type_p2sh": 56,
-            "bech32_prefix": null,
-            "blockchain_link": {
-                "type": "blockbook",
-                "url": ["https://blockbook.polispay.org"]
```
Comment (mroz22) [inline packages/connect-common/files/coins.json:1602]: this url does not even work anymore.

File: packages/components/src/components/logos/CoinLogo/coins.ts (hunk 1 above)
Comment (mroz22) [reply packages/components/src/components/logos/CoinLogo/coins.ts:18]: good for now. could be done anytime later.

**Comment by mroz22** [comment]
created a follow-up issue https://github.com/trezor/trezor-suite/issues/6599

File: suite-common/wallet-utils/src/__tests__/validation.test.ts [hunk 6]
```diff
@@ -79,6 +79,10 @@ describe('validation', () => {
         expect(isAddressValid('0xE37c0D48d68da5c5b14E5c1a9f1CFE802776D9FF0', 'trop')).toEqual(
             false,
         );
+        expect(isAddressValid('0x7de62F23453E9230cC038390901A9A0130105A3c', 'tgor')).toEqual(true);
```
Comment (mroz22) [inline suite-common/wallet-utils/src/__tests__/validation.test.ts]: in fact, this ~~throws error~~, console.errors an error so the tests here are not very useful :D see followup issue https://github.com/trezor/trezor-suite/issues/6599

**Review by mroz22 — CHANGES_REQUESTED** [review]
please add tgor also here 
- https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/connect/src/data/config.ts#L110

this is something related to invity part, might also need updatning?
https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/SendCryptoInput/SendCryptoSelect/index.tsx#L51 cc @yanascz 

optionally also in connect-explorer but it does not have that high priority so I leave it up to you
- https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/connect-explorer/src/data/methods/blockchain/index.ts#L19
- https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/connect-explorer/src/data/methods/ethereum/common.ts#L4
and few other places

**Comment by martinboehm** [comment]
@karliatto Please add 'TGOR' also to the invity file https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/SendCryptoInput/SendCryptoSelect/index.tsx?rgh-link-date=2022-10-19T10%3A59%3A49Z#L51. It is not actually that important as you anyway cannot do a real exchange of the testnet coins but it would be nice to have there for consistency.

**Comment by karliatto** [comment]
> @karliatto Please add 'TGOR' also to the invity file https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/suite/src/views/wallet/coinmarket/exchange/components/ExchangeForm/Inputs/SendCryptoInput/SendCryptoSelect/index.tsx?rgh-link-date=2022-10-19T10%3A59%3A49Z#L51. It is not actually that important as you anyway cannot do a real exchange of the testnet coins but it would be nice to have there for consistency.

@martinboehm what about https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/views/wallet/coinmarket/sell/components/SellForm/Inputs/CryptoInput/index.tsx#L200 ? it looks like the test coins are there as well but I am not familiarized wit that section.

**Comment by karliatto** [comment]
> please add tgor also here
> 
>     * https://github.com/trezor/trezor-suite/blob/feat/add-goerli/packages/connect/src/data/config.ts#L110

@mroz22 Is it required also to change the section `min` ? I understand this is for `supportedFirmware` and I am not sure if there is any implication that `tgor` is not supported in previous firmware.

File: suite-common/wallet-utils/src/__tests__/validation.test.ts (hunk 6 above)
Comment (karliatto) [reply suite-common/wallet-utils/src/__tests__/validation.test.ts]: I see, I removed that test, we can address it in the follow up issue. Thanks.

**Review by mroz22 — APPROVED** [review]

**Pull request merged**